dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java

```diff
@@ -390,21 +388,35 @@ public String extractFile(InputStream inputStream, String fileType) {
 
             Files.copy(inputStream, tempFile, StandardCopyOption.REPLACE_EXISTING);
 
-            zipFileSystem = FileSystems.newFileSystem(tempFile, (ClassLoader) null);
-            Path root = zipFileSystem.getPath("/");
-            Files.walk(root)
-                    .forEach(path -> {
-                        try {
-                            long fileSize = Files.size(path);
-                            if (Files.isDirectory(path)) {
-                                filePaths.add(path.toString().substring(1) + "/|" + fileSize );
-                            } else {
-                                filePaths.add(path.toString().substring(1) + "|" + fileSize );
-                            }
-                        } catch (IOException e) {
-                            e.printStackTrace();
+            if ("tar".equals(fileType)) {
+                try (InputStream fi = Files.newInputStream(tempFile);
+                     TarArchiveInputStream tis = new TarArchiveInputStream(fi)) {
+                    TarArchiveEntry entry;
+                    while ((entry = tis.getNextTarEntry()) != null) {
+                        if (entry.isDirectory()) {
+                            filePaths.add(entry.getName() + "/|" + entry.getSize());
+                        } else {
+                            filePaths.add(entry.getName() + "|" + entry.getSize());
                         }
-                    });
+                    }
+                }
+            } else {
+                zipFileSystem = FileSystems.newFileSystem(tempFile, (ClassLoader) null);
+                Path root = zipFileSystem.getPath("/");
+                Files.walk(root)
+                        .forEach(path -> {
+                            try {
```

Comment:
nested try catch  is that right?

Reply:
it should be effective for automatic closure of file